File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/DateTimeFormatComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;

import com.ekino.oss.jcv.core.JsonValueComparator;
import lombok.RequiredArgsConstructor;
import org.skyscreamer.jsonassert.ValueMatcherException;

@RequiredArgsConstructor
public class DateTimeFormatComparator implements JsonValueComparator<String> {

    private final DateTimeFormatter dateTimeFormatter;

    @Override
    public boolean hasCorrectValue(String actual, String expected) {
        try {
            //noinspection ResultOfMethodCallIgnored
            dateTimeFormatter.parse(actual);
            return true;
        } catch (DateTimeParseException e) {
            throw new ValueMatcherException(
                "Invalid date time format",
                e, expected, actual);
        }
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/DateTimeFormatComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeParseException

class DateTimeFormatComparator(private val dateTimeFormatter: DateTimeFormatter) : JsonValueComparator<String> {

  override fun hasCorrectValue(actual: String?, expected: String?): Boolean {
    try {
      dateTimeFormatter.parse(actual)
      return true
    } catch (e: DateTimeParseException) {
      throw ValueMatcherException("Invalid date time format", e, expected, actual)
    }
  }
}
